cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobContext.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.runtimejob;
+
+import org.apache.twill.api.TwillRunner;
+import org.apache.twill.api.TwillRunnerService;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+
+import java.util.Map;
+
+/**
+ * Runtime job context to provide information needed by the {@link RuntimeJob}.
+ */
+public interface RuntimeJobContext {
+  /**
+   * Returns a {@link DiscoveryService} for service announcement purpose.
+   */
+  DiscoveryService getDiscoveryService();
+
+  /**
+   * Returns a {@link DiscoveryServiceClient} for service discovery purpose.
+   */
+  DiscoveryServiceClient getDiscoveryServiceClient();
+
+  /**
+   * Returns a {@link TwillRunnerService} for running {@link TwillRunner}.
+   */
+  TwillRunnerService getTwillRunnerService();
+
+  /**
+   * Returns properties to be added to the CDAP configuration.
```

Comment:
don't really understand what this means. The implementation should set these in the CDAP conf? How is it populated?